eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/mutable/AbstractMutableBagIterable.java

```diff
@@ -303,7 +304,14 @@ public T max(Comparator<? super T> comparator)
             Function2<? super V, ? super T, ? extends V> nonMutatingAggregator)
     {
         MutableMap<K, V> map = UnifiedMap.newMap();
-        this.forEach(new NonMutatingAggregationProcedure<>(map, groupBy, zeroValueFactory, nonMutatingAggregator));
+        Procedure<T> nonMutatingAggregationProcedure = new NonMutatingAggregationProcedure<>(map, groupBy, zeroValueFactory, nonMutatingAggregator);
+        this.forEachWithOccurrences((each, occurrences) ->
+        {
+            for (int i = 0; i < occurrences; i++)
```

Comment:
You could reuse the method you move to NonMutatingAggregationProcedure.

Reply:
Reused the method from NonMutatingAggregationProcedure